### PR TITLE
Add virtual file system

### PR DESCRIPTION
**Pull request by LucasPickering**
Holy fuck this is a big one.

- Added a virtual file system implementation
  - The `vfs` module provides an abstraction layer between the DB and the web API, which means the API behave like a file system
- Added tests to the api crate that require DB access, so I had to update the CI to do that
- Added some funcs to the model types to make it easier to run queries on them
- Lots of other shit too

gl my dude

### CODE REVIEW

File: api/src/vfs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+    /// indicates a bug, hence why it panics in that case.
+    fn get_var(&self, var_name: &'static str) -> &str {
+        self.variables.get(var_name).expect("Unknown path variable")
+    }
+
+    fn set_if_var(&mut self, path_segment: &SegmentSpec, value: &'a str) {
+        if let SegmentSpec::Variable(var_name) = path_segment {
+            self.variables.insert(var_name, value);
+        }
+    }
```
Comment (JRMurr) [inline api/src/vfs/mod.rs]: confused on where `var_name` is coming from

Comment (JRMurr) [reply api/src/vfs/mod.rs]: nvm its like a match, but a little clean up would be cool

File: api/src/error.rs [hunk 2]
```diff
@@ -1,24 +1,43 @@
 use actix_web::{HttpResponse, ResponseError};
 use failure::Fail;
 
+pub type Result<T> = std::result::Result<T, ServerError>;
```
[the remark this answers is not in the document]
Comment (LucasPickering) [reply api/src/error.rs]: yeet, I don't really like that we re-use the name `Result` but that's the normal pattern used in Rust

File: api/src/vfs/mod.rs [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+                                SegmentSpec::Fixed("program.gdlk"),
+                                &ProgramSourceNodeHandler(),
+                            ),
+                        ],
+                    )],
+                ),
+            ],
+        )],
+    )],
+);
```
[the remark this answers is not in the document]
Comment (LucasPickering) [reply api/src/vfs/mod.rs]: like I said, I dont wanna

**Pull request merged**